### ISSUE TITLE
[BUG] Permissions are too high in `.github/workflows/dependent-issues.yml`

### ISSUE THREAD

**Issue by wwahammy** [houdiniproject/houdini#718]
## Description
For the best security, we should only give permissions to the workflow in `.github/workflows/dependent-issues.yml` for writing and reading issues and pull requests and nothing more.

## Possible Fix
Set the [permissions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#permissions) attribute to `none` for everything except `issues` and `pull_requests` which should be `write`.

**Comment by thiagolopess** [comment]
I have good news, @wwahammy. I forked your repo and started messing around with the permissions until I found what was the one that the workflow needed in order to work properly. Turns out it also needed write permission over the statuses. I'm gonna update the workflow with the change right away and open another PR.

![image](https://user-images.githubusercontent.com/48963026/139920473-d8d286e2-98a2-4ed9-8a81-6f59af405f4e.png)
